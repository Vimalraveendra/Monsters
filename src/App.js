        import React , {Component } from 'react';
        import CardList from './components/CardList/CardList';
        import './App.css';
        import  SearchBox  from './components/SearchBox/SearchBox';

        class App extends Component{
                constructor(){
                  super()
                  this.state={
                    monsters:[],
                    searchField:''
                  }
                }
          

          handleChange= e=>{
            this.setState({searchField:e.target.value})
          }
          componentDidMount(){
                   fetch('https://jsonplaceholder.typicode.com/users')
                  .then(response=>response.json())
                  .then(users=>this.setState({monsters:users}))
          }
        
          render(){
                  const {monsters, searchField} = this.state;
                  const filteredMonsters = monsters.filter(monster=>{
                   return monster.name.toLowerCase().includes(searchField.toLocaleLowerCase())
            })
          return (
                  
                      <div className="App">
                      <h1>Monsters Rolodex</h1>
                        <SearchBox 
                          placeholder="search monsters"
                         handleChange={this.handleChange}
                         />
                         <CardList   monsters ={filteredMonsters} />
                      </div>
          );
        }
        }
        export default App;
