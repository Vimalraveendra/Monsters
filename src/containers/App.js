import React, { Component } from "react";

import "./App.css";

import MainPage from "../components/MainPage/MainPage";

class App extends Component {
  constructor() {
    super();
    this.state = {
      monsters: [],
      searchField: "",
    };
  }

  handleChange = (e) => {
    this.setState({ searchField: e.target.value });
  };

  fetchCall = (fetch) => {
    return fetch("https://jsonplaceholder.typicode.com/users")
      .then((response) => response.json())
      .then((users) => this.setState({ monsters: users }));
  };

  render() {
    return (
      <MainPage
        monsters={this.state.monsters}
        searchField={this.state.searchField}
        fetchCall={this.fetchCall}
        handleChange={this.handleChange}
      />
    );
  }
}
export default App;
