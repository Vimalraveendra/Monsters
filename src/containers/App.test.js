import { shallow } from "enzyme";
import React from "react";
import App from "./App";

import fetchMock from "fetch-mock";

// import fetchMock from "fetch-mock";

let wrapper;
beforeEach(() => {
  wrapper = shallow(<App />);
});

it("should render the App component without crashing", () => {
  expect(wrapper).toMatchSnapshot();
});

describe("handleChange", () => {
  it("should handle the change and setstate after recieving the input", () => {
    const mockProps = {
      target: {
        value: "hel",
      },
    };

    const expectedAction = {
      monsters: [],
      searchField: "hel",
    };
    wrapper.instance().handleChange(mockProps);
    expect(wrapper.state()).toEqual(expectedAction);
  });
});

// describe("fetchAPI", () => {
//   it("should call fetchApi to get the monsters", async () => {
//     const mockProps = {
//       monsters: [
//         {
//           id: 123,
//           name: "hello",
//           email: "hello@gmail.com",
//         },
//       ],
//     };

//     fetchMock
//       .getOnce("https://jsonplaceholder.typicode.com/users", mockProps)
//       .then(() => {
//         expect(wrapper.state()).toEqual("hello");
//       });
//   });
// });

// describe("fetchAPI", () => {
//   it("should call fetchApi to get the monsters", async () => {
//     const mockProps = {
//       monsters: [
//         {
//           id: 123,
//           name: "hello",
//           email: "hello@gmail.com"
//         }
//       ]
//     };

//     fetchMock
//       .getOnce("https://jsonplaceholder.typicode.com/users", mockProps)
//       .then(() => {
//         expect(wrapper.state()).toEqual("hello");
//       });
//   });
// });
