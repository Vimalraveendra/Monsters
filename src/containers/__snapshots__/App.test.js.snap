// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the App component without crashing 1`] = `ShallowWrapper {}`;
